builtin_regexp.go

```diff
@@ -235,12 +241,24 @@ func compileRegexp(patternStr, flags string) (p *regexpPattern, err error) {
 		patternStr = convertRegexpToUtf16(patternStr)
 	}
 
-	re2Str, err1 := parser.TransformRegExp(patternStr)
+	var (
+		re2Str string
+		err1   error
+	)
+	if !dotAll {
+		re2Str, err1 = parser.TransformRegExp(patternStr)
+	} else {
+		err1 = parser.RegexpErrorIncompatible{}
+	}
 	if err1 == nil {
 		re2flags := ""
 		if multiline {
 			re2flags += "m"
 		}
+		if dotAll {
```

Comment:
dotAll cannot be true here because of the condition check at line 248. Also, replacing parser.Re2Dot with "." is wrong, consider if I have `"[^\r\n\u2028\u2029]"` as part of my regex, I don't want it to be implicitly replaced with a dot.

Reply:
My mistake, this code was my first solution, I forgot to delete it